src/com/teamcautionrobotics/robot2018/Harvester.java

```diff
@@ -0,0 +1,154 @@
+package com.teamcautionrobotics.robot2018;
+
+import com.teamcautionrobotics.misc2018.AbstractPIDSource;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import edu.wpi.first.wpilibj.VictorSP;
+
+public class Harvester {
+
+    public enum HarvesterAngle {
+        UP(90), AIMED(45), DOWN(0);
+
+        private static HarvesterAngle[] values = values();
+
+        public final double angle;
+
+        private HarvesterAngle(double angle) {
+            this.angle = angle;
+        }
+    }
+
+    private VictorSP grabber;
+    private VictorSP angularOptimizer;
+
+    private Encoder angularOptimizerEncoder;
+    private DigitalInput colorSensor;
+
+    private PIDController pidController;
+
+    public Harvester(int grabberChannel, int angularOptimizerChannel,
+            int angularOptimizerEncoderChannelA, int angularOptimizerEncoderChannelB,
+            int colorSensorChannel, double Kp, double Ki, double Kd) {
+
+        grabber = new VictorSP(grabberChannel);
+        angularOptimizer = new VictorSP(angularOptimizerChannel);
+
+        colorSensor = new DigitalInput(colorSensorChannel);
+
+        angularOptimizerEncoder =
+                new Encoder(angularOptimizerEncoderChannelA, angularOptimizerEncoderChannelB);
+        angularOptimizerEncoder.setDistancePerPulse(360 / 1024);
```

Comment:
This scaling factor will be wrong, since there are gears with non-unity ratios between the encoder and the grabber.